### PR TITLE
[NO-JIRA]: Decision for versioning

### PR DESCRIPTION
**Pull request by ojcurtis**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: decisions/versioning-rules.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!-- Copy this to make your own decisions. -->
+
+# Versioning approach for component releases
+
+## TL;DR
+<!-- Only if long enough to merit one. -->
```
Comment (shaundon) [inline decisions/versioning-rules.md]: You can drop this section as it's empty.

File: decisions/versioning-rules.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Decision
+When a new Backpack component is to be released the approach is to use semver to manage versions which are based on the types of changes that have been made to each component. To ensure that changes made don't cause unexpected outages or issues throughout the applications the following reasoning should be followed.
```
Comment (shaundon) [inline decisions/versioning-rules.md]: Can you make the word `semver` be a link to https://semver.org/? E.g. [semver](https://semver.org/).

File: decisions/versioning-rules.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+### Major
+For Major releases these would be changes that do the following:
```
Comment (shaundon) [inline decisions/versioning-rules.md]:
```suggestion
For major releases these would be changes that do the following:
```

File: decisions/versioning-rules.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- If changes are not backwards compatible.
+- Breaking changes to components such as updating the API or visual changes.
+- Changes to tokens such as changes to colours or changes to values such as padding/margins.
+- Deprecation to tokens, components or API.
+- Adding new functionality to a component.
+
+### Minor
+For Minor releases these would be changes that do the following:
```
Comment (shaundon) [inline decisions/versioning-rules.md]:
```suggestion
For minor releases these would be changes that do the following:
```

File: decisions/versioning-rules.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+## TL;DR
+<!-- Only if long enough to merit one. -->
+
+## Decision
+When a new Backpack component is to be released the approach is to use semver to manage versions which are based on the types of changes that have been made to each component. To ensure that changes made don't cause unexpected outages or issues throughout the applications the following reasoning should be followed.
+
+### Major
+For Major releases these would be changes that do the following:
+- If changes are not backwards compatible.
+- Breaking changes to components such as updating the API or visual changes.
```
Comment (shaundon) [inline decisions/versioning-rules.md]: I think we need to spend a bit more time figuring out what a breaking visual change consists of. That's probably outside the scope of this PR but we figure it out soon nonetheless.

File: decisions/versioning-rules.md (hunk 2 above)
Comment (georgegillams) [inline decisions/versioning-rules.md]:
```suggestion
When we release Backpack packages, we use [semver](https://semver.org/) to ensure that changes made don't cause unexpected issues. The following reasoning should be followed.
```

File: decisions/versioning-rules.md (hunk 3 above)
Comment (georgegillams) [inline decisions/versioning-rules.md]:
```suggestion
A change is `major` if:
```

File: decisions/versioning-rules.md (hunk 4 above)
Comment (georgegillams) [inline decisions/versioning-rules.md]:
```suggestion
A change is `minor` if:
```

File: decisions/versioning-rules.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+### Minor
+For Minor releases these would be changes that do the following:
+- Changes where the updates are only in the Backpack component itself and not used publicly.
+- Added new functionality to a component that didn't exist before and are not mandatory meaning there would be no impact on release.
+- Adding a new icon that is not in used yet.
+- Adding a new component to the Backpack library.
+
+### Patch
+- Only a dependency that a component uses needs an upgrade to improve functionality in the component.
```
Comment (georgegillams) [inline decisions/versioning-rules.md]:
```suggestion
A change is `minor` if:
- Only a dependency that a component uses needs an upgrade to improve functionality in the component.
```

File: decisions/versioning-rules.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+- Bug fixes.
+- Quality fixes such as `flow`, `eslint` or `prettier`.
+
+## Thinking
+The thinking behind this will align contributors on how to decide a release version for component development.
```
Comment (georgegillams) [inline decisions/versioning-rules.md]:
```suggestion
[You can find out more about the benefits of semver on semver.org](https://semver.org/#why-use-semantic-versioning)
```

File: decisions/versioning-rules.md (hunk 5 above)
Comment (ojcurtis) [reply decisions/versioning-rules.md]: I guess we could look at creating a notes section to outline and detail what a visual change is?

File: decisions/versioning-rules.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<!-- Copy this to make your own decisions. -->
+
+# Versioning approach for component releases
+
+## Decision
+When we release Backpack packages, we use [semver](https://semver.org/) to ensure that changes made don't cause unexpected issues. The following reasoning should be followed.
+
+### Major
+A change is `major` if:
+- If changes are not backwards compatible.
```
Comment (georgegillams) [inline decisions/versioning-rules.md]: Just one more tiny thing from me:
```suggestion
- Changes are not backwards compatible.
```

**Pull request merged**